src/client/utils/date/date.ts

```diff
@@ -59,3 +66,74 @@ export function formatTimeRange(timeRange: TimeRange, timezone: Timezone, displa
 
   return formatted;
 }
+
+// calendar utils
+
+export function monthToWeeks(startDate: Date, tzString: string): Date[][] {
+  const weeks: Date[][] = [];
+  var week: Date[] = [];
+  var firstDayOfMonth = wallTimePreciseToMonth(startDate, tzString);
+  for (var i = 1; i <= getCountDaysInMonth(firstDayOfMonth); i++) {
+    var activeDay = wallTimePreciseToDay(startDate, i, tzString);
+    // this assumes week starts on sunday..
+    if (activeDay.getDay() === 0 && week.length > 0) {
+      weeks.push(week);
+      week = [];
+    }
+    week.push(activeDay);
+  }
+  // push last week
+  if (week.length > 0) weeks.push(week);
+  return weeks;
+}
+
+export function getCountDaysInMonth(date: Date): number {
+  if (!isDate(date)) return 0;
+  return new Date(date.getFullYear(), date.getMonth() + 1, 0).getDate();
+}
+
+export function prependDays(timezone: Timezone, weekPrependTo: Date[], countPrepend: number): Date[] {
+  for (var i = 0; i < countPrepend; i++) {
+    var firstDate = weekPrependTo[0];
+    var shiftedDate = day.shift(firstDate, timezone, - 1);
+    weekPrependTo.unshift(shiftedDate);
+  }
+  return weekPrependTo;
+}
+
+export function appendDays(timezone: Timezone, weekAppendTo: Date[], countAppend: number): Date[] {
+  for (var i = 0; i < countAppend; i++) {
+    var lastDate = weekAppendTo[weekAppendTo.length - 1];
+    var shiftedDate = day.shift(lastDate, timezone, 1);
+    weekAppendTo.push(shiftedDate);
+  }
+  return weekAppendTo;
+}
+
+export function daysEqualWallTime(d1: Date, d2: Date, timezone: string): boolean {
```

Comment:
This function's name is not super explicit. Maybe something like `wallTimeDaysAreEqual` would be better ? Or maybe it's just me ^^